src/components/SideBar.tsx

```diff
@@ -112,24 +127,30 @@ function SideBar({
 			)}
 			<AddCourseButtonContainer>
 				<ActionButton
-					onClick={openModal}
+					onClick={openCourseModal}
 					variant="secondary"
 					disabled={!currentSemester}
 				>
 					Add course
 				</ActionButton>
 			</AddCourseButtonContainer>
-			{modalOpen && (
+			{courseModalOpen && (
 				<AddCourseModal
-					handleSubmit={handleSubmit}
+					handleSubmit={handleCourseSubmit}
 					handleClose={handleModalClose}
 				/>
 			)}
 			<AddSemesterButtonContainer>
-				<ActionButton onClick={openModal} variant="secondary">
+				<ActionButton onClick={openSemesterModal} variant="secondary">
 					Add Semester
 				</ActionButton>
 			</AddSemesterButtonContainer>
+			{semesterModalOpen && (
+				<AddSemesterModal
+					handleSubmit={handleSemesterSubmit}
+					handleClose={handleModalClose}
+				/>
+			)}
 			<PickSemesterButtonContainer>
 				<SemesterPicker
 					semesters={[Fall2020, Fall2021]} //Replace mock data
```

Comment:
Previously, there was an clash between the modals so I made some changes in here just to get the semester modal working (clicking the 'add semester' button would open the course modal since there was no distinction between them in the code)